src/WingetCreateCLI/Commands/NewCommand.cs

```diff
@@ -237,10 +239,136 @@ private static void PromptRequiredProperties<T>(T manifest, VersionManifest vers
                         continue;
                     }
 
-                    var currentValue = property.GetValue(manifest);
-                    var result = PromptProperty(manifest, currentValue, property.Name);
-                    property.SetValue(manifest, result);
-                    Logger.Trace($"Property [{property.Name}] set to the value [{result}]");
+                    PromptHelper.PromptProperty(manifest, property.Name, property.GetValue(manifest));
+                    Logger.Trace($"Property [{property.Name}] set to the value [{property.GetValue(manifest)}]");
+                }
+            }
+        }
+
+        /// <summary>
+        /// Displays all installers from an Installer manifest as a selection menu.
+        /// </summary>
+        private static void DisplayInstallersAsMenuSelection(InstallerManifest installerManifest)
+        {
+            Console.Clear();
+
+            while (true)
+            {
+                List<string> selectionList = GenerateInstallerSelectionList(installerManifest.Installers);
+                var selectedItem = Prompt.Select(Resources.SelectInstallerToEdit_Message, selectionList);
+
+                if (selectedItem == Resources.None_MenuItem)
+                {
+                    break;
+                }
+                else if (selectedItem == Resources.AllInstallers_MenuItem)
+                {
+                    Installer installerCopy = new Installer();
+                    PromptHelper.PromptPropertiesWithMenu(installerCopy, Resources.None_MenuItem);
+                    ApplyChangesToIndividualInstallers(installerCopy, installerManifest.Installers);
+                }
+                else if (selectedItem == Resources.DisplayPreview_MenuItem)
+                {
+                    Console.Clear();
+                    Console.WriteLine();
+                    Logger.InfoLocalized(nameof(Resources.DisplayPreviewOfSelectedInstaller_Message));
+                    var serializer = Serialization.CreateSerializer();
+                    string installerString = serializer.Serialize(installerManifest);
+                    Console.WriteLine(installerString);
+                    Console.WriteLine();
+                }
+                else
+                {
+                    Installer selectedInstaller = MatchMenuSelectionToInstaller(selectedItem, installerManifest);

```

Comment:
Sorry, I didn't think of this before, but what if instead of trying to reverse-engineer the string to get the Installer, you just create a temporary Dictionary of tuplestring -> Installer, and then after the user selects one, you can look it up in the dictionary to get the Installer? Seems like that'd be less error-prone and easier to maintain

Reply:
Good idea, changed to suggestion

Reply:
Nice, looks a lot better!